### PR TITLE
Add documention to Job#moveToCompleted/Failed

### PR DESCRIPTION
**Pull request by aleccool213**
- Also add multiple repo to PATTERN.md

Issues which reference this pattern:

https://github.com/OptimalBits/bull/issues/790
https://github.com/OptimalBits/bull/issues/208

### CODE REVIEW

**Comment by aleccool213** [comment]
Ran the tests with node 6 and they all passed on my mac :(

**Comment by manast** [comment]
Thanks for the PR. Although this pattern may seem to work, there is at least one problem with it, currently, moveToCompleted will fetch the next job if there are jobs waiting, this is to avoid an extra roundtrip to redis to fetch the next job in the scenario with a busy queue: https://github.com/OptimalBits/bull/blob/master/lib/commands/moveToFinished-6.lua#L58-L76

**Comment by aleccool213** [comment]
@manast I have documented that behaviour in this [pr](https://github.com/OptimalBits/bull/pull/1017/files#diff-eb120e74eba90da6132666ceb0405bf9R705) and in the corresponding typings [pr](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/27816/files). We actually wanted to do this anyways in our implementation of the pattern so this "problem" is actually a feature for us 🤣

**Comment by aleccool213** [comment]
@manast Neat!

Cool, I will add some unit tests.

**Comment by aleccool213** [comment]
@manast new commits are up for unit testing #moveToCompleted

File: REFERENCE.md [hunk 1]
```diff
@@ -694,6 +696,27 @@ Returns a promise that resolves or rejects when the job completes or fails.
...
 ---
 
+### Job#moveToCompleted
+
+```ts
+moveToCompleted(returnValue, ignoreLock): Promise<string[]>
+```
+
+Moves a job to the completed queue. Pulls a job from 'waiting' to 'active' and returns a promise with it's id.
+
```
Comment (manast) [inline REFERENCE.md]: It actually does not return an Id, but a tuple containing the next job data and the next job id (as commented above) or null, if no more jobs left in the waiting list. So maybe we should declare it like this:
```typescript
moveToCompleted(returnValue, ignoreLock): Promise<[JobData, JobId] | null>
```

File: REFERENCE.md [hunk 2]
```diff
@@ -694,6 +696,27 @@ Returns a promise that resolves or rejects when the job completes or fails.
...
+---
+
+### Job#moveToFailed
+
+```ts
+moveToFailed(errorInfo, ignoreLock): Promise<void>
```
Comment (manast) [inline REFERENCE.md]: it does return the same as moveToCompleted

File: REFERENCE.md (hunk 1 above)
Comment (aleccool213) [reply REFERENCE.md]: 👍

File: PATTERNS.md [hunk 3]
```diff
@@ -197,3 +198,44 @@ myQueue.add({msg: 'Specific Error'}, {
...
+2. Pulling a job from 'waiting' and moving it to 'active'.
+
+```typescript
+const job: Job = await queue.getNextJob();
+```
+
+3. (Optional) Move the job to the 'failed' queue if something goes wrong.
+
+```typescript
+await job.moveToFailed(
```
[the remark this answers is not in the document]
Comment (aleccool213) [reply PATTERNS.md]: 👍

File: PATTERNS.md [hunk 4]
```diff
@@ -197,3 +198,44 @@ myQueue.add({msg: 'Specific Error'}, {
...
+);
+```
+
+4. Move the job to the 'completed' queue.
+
+```typescript
+const newJob = await job.moveToCompleted("succeeded", true);
+```
+
+Then you can easily wrap `bull` in an API for use with external systems.
```
[the remark this answers is not in the document]
Comment (aleccool213) [reply PATTERNS.md]: 👍

File: PATTERNS.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (aleccool213) [reply PATTERNS.md]: Hmmm in my testing, I have a job in the `waiting` queue and this function consistently doesn't return me the tuple you outlined :(

I will add the documentation anyways tho but thought I would let you know.

[the remark this answers is not in the document]
Comment (aleccool213) [reply PATTERNS.md]: @manast ^

**Comment by aleccool213** [comment]
@manast Made all the requested changes. Please have at it again 😄

File: PATTERNS.md [hunk 5]
```diff
@@ -197,3 +198,52 @@ myQueue.add({msg: 'Specific Error'}, {
  }
 });
 ```
+
+Async multiple-repo job processor
```
Comment (manast) [inline PATTERNS.md]: I am a bit unsure about the title, maybe it is not so clear. So the pattern is actually about "manually fetching jobs" from the queue, maybe that would be better, then the user may use it for many different scenarios.

File: PATTERNS.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (manast) [reply PATTERNS.md]: yes, you are right, it is not returning the next job, but I will add it in the next version for consistency. In any case it is just an optimization, if the user does not get the next job he will fetch it manually.

**Comment by manast** [comment]
I will merge this PR during the week. Good job!

File: PATTERNS.md (hunk 5 above)
Comment (aleccool213) [reply PATTERNS.md]: totally

**Pull request merged**